pkg/api/unversioned.go

```diff
@@ -22,16 +22,64 @@ import (
 
 // This file contains API types that are unversioned.
 
-// APIVersions lists the api versions that are available, to allow
-// version negotiation. APIVersions isn't just an unnamed array of
-// strings in order to allow for future evolution, though unversioned
+// APIVersions lists the versions that are available, to allow clients to
+// discover the API at /api, which is the root path of the legacy v1 API.
 type APIVersions struct {
+	// versions are the api versions that are available.
 	Versions []string `json:"versions"`
 }
 
+// APIGroupList is a list of APIGroup, to allow clients to discover the API at
+// /apis.
+type APIGroupList struct {
+	// groups is a list of APIGroup.
+	Groups []APIGroup `json:"groups"`
+}
+
+// APIGroup contains the name, the supported versions, and the preferred version
+// of a group.
+type APIGroup struct {
+	// name is the name of the group.
+	Name string `json:"name"`
+	// groupVersions are the versions supported in this group.
+	GroupVersions []GroupVersion `json:"groupVersions"`
```

Comment:
Yes, you convinced me. The "preferredGroupVersion" sounds redundant. For consistency, I will rename this field to Versions.